### PR TITLE
FEValuesViews output traits

### PR DESCRIPTION
**Pull request by luca-heltai**
Extracted from #7806.

### CODE REVIEW

File: include/deal.II/fe/fe_values_views_traits.h [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    /**
+     * Partial specialization for FEValuesExtractors::SymmetricTensor<2>.
+     */
+    template <int dim, int spacedim, typename Number>
+    struct OutputTypeTrait<dim,
+                           spacedim,
+                           Number,
+                           FEValuesExtractors::SymmetricTensor<2>>
+    {
+      using OutputType = typename FEValuesViews::
```
Comment (jppelteret) [inline include/deal.II/fe/fe_values_views_traits.h]: Based on your tests, shouldn't this be `value_type`?

File: include/deal.II/fe/fe_values_views_traits.h [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+DEAL_II_NAMESPACE_OPEN
+
+namespace FEValuesViews
+{
+  namespace
+  {
+    /**
+     * A trait for the Output type required when extracting an
+     * FEValuesView from FEValuesBase using an object in the namespace
+     * FEValuesExtractors. The indirection is needed here, because the c++11
```
Comment (jppelteret) [inline include/deal.II/fe/fe_values_views_traits.h]: c++11 -> C++11

**Comment by bangerth** [comment]
I think I haven't quite understood yet what you need this for. Is this just an elaborate way to translate from an `FEValuesExtractor` to an `FEValuesView`? Because if you had that, say if there was something like

```
template <typename Extractor>
FEValuesViews::ViewFromExtractor {
  using type = ...;
};
```
then you could just use
```
  FEValuesView::ViewFromExtractor<...>::type::OutputType<...>
```
(possibly plus some `typename` and `template` in a couple of places.

I think this would be a more interesting solution because it might apply in other places as well.

**Comment by luca-heltai** [comment]
@bangerth , would this work for you?

**Comment by bangerth** [comment]
Yes, this is excellent. Thank you!

**Pull request merged**